cql3/expr/expression.hh

```diff
@@ -183,14 +191,53 @@ struct function_call {
 
 struct cast {
     nested_expression arg;
-    cql3_type type;
+    std::variant<cql3_type, shared_ptr<cql3_type::raw>> type;
 };
 
 struct field_selection {
     nested_expression structure;
     shared_ptr<column_identifier_raw> field;
 };
 
+struct null {
+};
+
+struct bind_variable {
+    struct scalar {};
+    struct scalar_in {};
+    struct tuple {};
+    struct tuple_in {};
+    // FIXME: infer shape from expression rather than from grammar
+    std::variant<scalar, scalar_in, tuple, tuple_in> shape;
```

Comment:
Variant of empty structs seems strange, why not an `enum`?

Reply:
I thought we might later have need for a real variant. Instead, it will probably disappear (and shape will be inferred using types instead of the grammar). I'll change it to enum.

Reply:
Fixed